Command key on macOS remains activated after window changes (cmd+TAB)

Since upgrade stable versions from 0.4.0 to 0.4.1 I have experienced that the command key seems to remain in a activated state when changing windows with cmd+TAB.

The result is then either unresponsive keys until cmd is pressed and released again, or when hitting keys like Q or H, the window quits or hides accordingly.

Also, it seems be reproducible consistently when you cmd+TAB to an "empty window" (i.e. on macOS some programs can run without graphical window).

### System
OS: macOS
Version: alacritty 0.4.1 (4d9253b)

when applying that patch at that commit, I first get
```
error[E0531]: cannot find tuple struct or tuple variant `ModifiersChanged` in this scope
   --> alacritty/src/event.rs:588:21
    |
588 |                     ModifiersChanged(modifiers) => {
    |                     ^^^^^^^^^^^^^^^^ not found in this scope
    |
help: possible candidate is found in another module, you can import it into scope
    |
2   | use glutin::event::DeviceEvent::ModifiersChanged;
    |
```
then after adding the import it still fails with
```
error[E0308]: mismatched types
   --> alacritty/src/event.rs:589:21
    |
518 |                 match event {
    |                       ----- this match expression has type `winit::event::WindowEvent<'_>`
...
589 |                     ModifiersChanged(modifiers) => {
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `winit::event::WindowEvent`, found enum `winit::event::DeviceEvent`
```

Is there a simple fix?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -69,6 +69,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Low resolution mouse cursor and decorations on HiDPI Wayland outputs
 - Decorations visible when in fullscreen on Wayland
 - Window size not persisted correctly after fullscreening on macOS
+- Crash on startup with some locales on X11
 
 ### Removed
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -66,7 +66,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Incorrect modifiers tracking on macOS, leading to 'sticky' modifiers
 - Crash when starting on Windows with missing dark mode support
 - Variables `XCURSOR_THEME` and `XCURSOR_SIZE` ignored on Wayland
-- Low resolution mouse cursor and decorations on HiDPI Wayland monitors
+- Low resolution mouse cursor and decorations on HiDPI Wayland outputs
 - Decorations visible when in fullscreen on Wayland
 - Window size not persisted correctly after fullscreening on macOS
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -63,7 +63,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Selection invisible when starting above viewport and ending below it
 - Clipboard not working after TTY switch on Wayland
 - Crash when pasting non UTF-8 string advertised as UTF-8 string on Wayland
-- Incorrect modifiers tracking on macOS, leading to 'sticky' modifiers
+- Incorrect modifiers tracking on X11 and macOS, leading to 'sticky' modifiers
 - Crash when starting on Windows with missing dark mode support
 - Variables `XCURSOR_THEME` and `XCURSOR_SIZE` ignored on Wayland
 - Low resolution mouse cursor and decorations on HiDPI Wayland outputs

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -48,7 +48,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Expansion of block-selection on partially selected full-width glyphs
 - Minimize action only works with decorations on macOS
 - Window permanently vanishing after hiding on macOS
-- Modifier key combinations like `Ctrl + Q` not generating characters on macOS
 - Handling of URLs with single quotes
 - Parser reset between DCS escapes
 - Parser stopping at unknown DEC private modes/SGR character attributes
@@ -64,6 +63,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Selection invisible when starting above viewport and ending below it
 - Clipboard not working after TTY switch on Wayland
 - Crash when pasting non UTF-8 string advertised as UTF-8 string on Wayland
+- Incorrect modifiers tracking on macOS, leading to 'sticky' modifiers
+- Crash when starting on Windows with missing dark mode support
+- Variables `XCURSOR_THEME` and `XCURSOR_SIZE` ignored on Wayland
+- Low resolution mouse cursor and decorations on HiDPI Wayland monitors
+- Decorations visible when in fullscreen on Wayland
+- Window size not persisted correctly after fullscreening on macOS
 
 ### Removed
 

--- a/alacritty/Cargo.toml
+++ b/alacritty/Cargo.toml
@@ -18,7 +18,7 @@ fnv = "1"
 serde = { version = "1", features = ["derive"] }
 serde_yaml = "0.8"
 serde_json = "1"
-glutin = { version = "0.23.0", features = ["serde"] }
+glutin = { version = "0.24.0", features = ["serde"] }
 notify = "4"
 libc = "0.2"
 unicode-width = "0.1"

--- a/alacritty/src/event.rs
+++ b/alacritty/src/event.rs
@@ -11,9 +11,7 @@ use std::sync::Arc;
 use std::time::Instant;
 
 use glutin::dpi::PhysicalSize;
-use glutin::event::{
-    ElementState, Event as GlutinEvent, ModifiersState, MouseButton, WindowEvent,
-};
+use glutin::event::{ElementState, Event as GlutinEvent, ModifiersState, MouseButton, WindowEvent};
 use glutin::event_loop::{ControlFlow, EventLoop, EventLoopProxy, EventLoopWindowTarget};
 use glutin::platform::desktop::EventLoopExtDesktop;
 use log::{debug, info, warn};
@@ -553,7 +551,9 @@ impl<N: Notify + OnResize> Processor<N> {
                         processor.mouse_input(state, button);
                         processor.ctx.terminal.dirty = true;
                     },
-                    WindowEvent::ModifiersChanged(modifiers) => processor.modifiers_input(modifiers),
+                    WindowEvent::ModifiersChanged(modifiers) => {
+                        processor.modifiers_input(modifiers)
+                    },
                     WindowEvent::CursorMoved { position, .. } => {
                         let (x, y) = position.into();
                         let x = limit(x, 0, processor.ctx.size_info.width as i32);

--- a/alacritty/src/event.rs
+++ b/alacritty/src/event.rs
@@ -12,7 +12,7 @@ use std::time::Instant;
 
 use glutin::dpi::PhysicalSize;
 use glutin::event::{
-    DeviceEvent, ElementState, Event as GlutinEvent, ModifiersState, MouseButton, WindowEvent,
+    ElementState, Event as GlutinEvent, ModifiersState, MouseButton, WindowEvent,
 };
 use glutin::event_loop::{ControlFlow, EventLoop, EventLoopProxy, EventLoopWindowTarget};
 use glutin::platform::desktop::EventLoopExtDesktop;
@@ -553,6 +553,7 @@ impl<N: Notify + OnResize> Processor<N> {
                         processor.mouse_input(state, button);
                         processor.ctx.terminal.dirty = true;
                     },
+                    WindowEvent::ModifiersChanged(modifiers) => processor.modifiers_input(modifiers),
                     WindowEvent::CursorMoved { position, .. } => {
                         let (x, y) = position.into();
                         let x = limit(x, 0, processor.ctx.size_info.width as i32);
@@ -603,13 +604,9 @@ impl<N: Notify + OnResize> Processor<N> {
                     | WindowEvent::Moved(_) => (),
                 }
             },
-            GlutinEvent::DeviceEvent { event, .. } => {
-                if let DeviceEvent::ModifiersChanged(modifiers) = event {
-                    processor.modifiers_input(modifiers);
-                }
-            },
             GlutinEvent::Suspended { .. }
             | GlutinEvent::NewEvents { .. }
+            | GlutinEvent::DeviceEvent { .. }
             | GlutinEvent::MainEventsCleared
             | GlutinEvent::RedrawEventsCleared
             | GlutinEvent::Resumed

